API Explorer not working on Windows 11

<!--
Please DO NOT ERASE this template, but fill in the details as requested. This will ensure your issue is properly filed and will be considered. Otherwise, we may reserve the right to close it without further action.
-->

## The problem

<!--
When I select "developer tools-> API Explorer" then I only get an empty screen.
![image](https://github.com/openhab/openhab-webui/assets/165837628/87ec21a3-a218-4b5e-b6ff-d08eef8d8300)


-->

## Expected behavior

<!--
API Explorer should show up
-->

## Steps to reproduce

<!--
Problem is always there, never used the API explorer since openHAB 2, so I don't know if it worked correctly on previous versions of openHAB 3&4.
-->

1. 


## Your environment

<!--
runtimeInfo:
  version: 4.1.2
  buildString: Release Build
locale: en-NL
systemInfo:
  configFolder: C:\openHAB\conf
  userdataFolder: C:\openHAB\userdata
  logFolder: C:\openHAB\userdata\logs
  javaVersion: 17.0.8.1
  javaVendor: Azul Systems, Inc.
  javaVendorVersion: Zulu17.44+53-CA
  osName: Windows 11
  osVersion: "10.0"
  osArchitecture: amd64
  availableProcessors: 4
  freeMemory: 193523688
  totalMemory: 448790528
  uptime: 564
  startLevel: 70
addons:
  - automation-jsscripting
  - binding-astro
  - binding-evohome
  - binding-exec
  - binding-fsinternetradio
  - binding-ipcamera
  - binding-logreader
  - binding-mqtt
  - binding-nanoleaf
  - binding-shelly
  - binding-tr064
  - binding-zigbee
  - binding-zwave
  - misc-openhabcloud
  - persistence-rrd4j
  - transformation-jsonpath
  - transformation-regex
  - ui-basic
  - ui-habpanel
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: false
    macos: false
    windows: true
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    os: windows
    pixelRatio: 1
    prefersColorScheme: dark
  isSecureContext: true
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: 8
    hardwareConcurrency: 4
    language: en
    languages:
      - en
      - nl-NL
      - de
      - nl
      - en-US
    onLine: true
    platform: Win32
  screen:
    width: 1600
    height: 900
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: dark
    filled: true
    pageTransitionAnimation: default
    bars: light
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
  userAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML,
    like Gecko) Chrome/123.0.0.0 Safari/537.36
timestamp: 2024-04-03T06:20:16.935Z

-->

```yaml

```

## Browser console

<!--
app.ac598e3ce6e035bd1153.js:7 
        
        
       POST http://localhost/rest/auth/token 400 (Bad Request)
s @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
s.promise @ app.ac598e3ce6e035bd1153.js:7
postPlain @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:23
refreshAccessToken @ app.ac598e3ce6e035bd1153.js:23
onPageAfterIn @ 15.app.ac598e3ce6e035bd1153.js:1
Ue @ app.ac598e3ce6e035bd1153.js:7
n @ app.ac598e3ce6e035bd1153.js:7
Ue @ app.ac598e3ce6e035bd1153.js:7
e.$emit @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
t.a @ app.ac598e3ce6e035bd1153.js:7
dispatchEvent @ app.ac598e3ce6e035bd1153.js:23
onPageAfterIn @ app.ac598e3ce6e035bd1153.js:23
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
value @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
value @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
value @ app.ac598e3ce6e035bd1153.js:7
value @ app.ac598e3ce6e035bd1153.js:7
q @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
o @ app.ac598e3ce6e035bd1153.js:1
c @ app.ac598e3ce6e035bd1153.js:1
app.ac598e3ce6e035bd1153.js:23 Bad Request
developer/api-explorer:1 Uncaught (in promise) Bad Request
Promise.then (async)
onPageAfterIn @ 15.app.ac598e3ce6e035bd1153.js:1
Ue @ app.ac598e3ce6e035bd1153.js:7
n @ app.ac598e3ce6e035bd1153.js:7
Ue @ app.ac598e3ce6e035bd1153.js:7
e.$emit @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
t.a @ app.ac598e3ce6e035bd1153.js:7
dispatchEvent @ app.ac598e3ce6e035bd1153.js:23
onPageAfterIn @ app.ac598e3ce6e035bd1153.js:23
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
value @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
value @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
value @ app.ac598e3ce6e035bd1153.js:7
value @ app.ac598e3ce6e035bd1153.js:7
q @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
(anonymous) @ app.ac598e3ce6e035bd1153.js:7
o @ app.ac598e3ce6e035bd1153.js:1
c @ app.ac598e3ce6e035bd1153.js:1

-->

```txt

```

## Browser network traffic

<!--
![image](https://github.com/openhab/openhab-webui/assets/165837628/bfba971d-db5e-4b59-a6e4-e16e4f197c65)
![image](https://github.com/openhab/openhab-webui/assets/165837628/910e2f89-3362-4b79-b33c-46d2c8431012)

-->

## Additional information

<!--
Looks like an authentication issue,  openHab log file below:
`2024-04-03 08:23:57.368 [ERROR] [.io.rest.auth.internal.TokenResource] - Error while authenticating
java.lang.NullPointerException: Cannot invoke "org.openhab.core.auth.UserSession.getRefreshToken()" because "s" is null
	at org.openhab.core.io.rest.auth.internal.TokenResource.lambda$7(TokenResource.java:382) ~[?:?]
	at java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90) ~[?:?]
	at java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1602) ~[?:?]
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129) ~[?:?]
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230) ~[?:?]
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:632) ~[?:?]
	at org.openhab.core.io.rest.auth.internal.TokenResource.lambda$6(TokenResource.java:382) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178) ~[?:?]
	at java.util.HashMap$KeySpliterator.tryAdvance(HashMap.java:1728) ~[?:?]
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129) ~[?:?]
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.findAny(ReferencePipeline.java:652) ~[?:?]
	at org.openhab.core.io.rest.auth.internal.TokenResource.processRefreshTokenGrant(TokenResource.java:383) ~[?:?]
	at org.openhab.core.io.rest.auth.internal.TokenResource.getToken(TokenResource.java:129) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179) ~[bundleFile:3.6.2]
	at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96) ~[bundleFile:3.6.2]
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:201) ~[bundleFile:3.6.2]
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:104) ~[bundleFile:3.6.2]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59) ~[bundleFile:3.6.2]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96) ~[bundleFile:3.6.2]
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:307) ~[bundleFile:3.6.2]
	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121) ~[bundleFile:3.6.2]
	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:265) ~[bundleFile:3.6.2]
	at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234) ~[bundleFile:3.6.2]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208) ~[bundleFile:3.6.2]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160) ~[bundleFile:3.6.2]
	at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:225) ~[bundleFile:3.6.2]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:304) ~[bundleFile:3.6.2]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doPost(AbstractHTTPServlet.java:217) ~[bundleFile:3.6.2]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:517) ~[bundleFile:4.0.4]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:279) ~[bundleFile:3.6.2]
	at org.ops4j.pax.web.service.spi.servlet.OsgiInitializedServlet.service(OsgiInitializedServlet.java:102) ~[bundleFile:?]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1656) ~[bundleFile:9.4.52.v20230823]
	at org.ops4j.pax.web.service.spi.servlet.OsgiFilterChain.doFilter(OsgiFilterChain.java:100) ~[bundleFile:?]
	at org.ops4j.pax.web.service.jetty.internal.PaxWebServletHandler.doHandle(PaxWebServletHandler.java:320) ~[bundleFile:?]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:600) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1440) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:505) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1355) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:234) ~[bundleFile:9.4.52.v20230823]
	at org.ops4j.pax.web.service.jetty.internal.PrioritizedHandlerCollection.handle(PrioritizedHandlerCollection.java:96) ~[bundleFile:?]
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:722) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.Server.handle(Server.java:516) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:487) ~[bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:732) [bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:479) [bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277) [bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) [bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) [bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338) [bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315) [bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173) [bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131) [bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409) [bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883) [bundleFile:9.4.52.v20230823]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034) [bundleFile:9.4.52.v20230823]
	at java.lang.Thread.run(Thread.java:833) [?:?]
`
-->

--- a/bundles/org.openhab.ui/web/package.json
+++ b/bundles/org.openhab.ui/web/package.json
@@ -86,6 +86,7 @@
     "@mit-app-inventor/blockly-plugin-workspace-multiselect": "^1.0.2",
     "@replit/codemirror-indentation-markers": "^6.5.3",
     "@replit/codemirror-vim": "^6.3.0",
+    "@types/swagger-ui-dist": "^3.30.6",
     "@uiw/codemirror-theme-gruvbox-dark": "^4.25.4",
     "@vue-leaflet/vue-leaflet": "^0.10.1",
     "blockly": "^11.2.2",

--- a/bundles/org.openhab.ui/web/src/pages/developer/api-explorer.vue
+++ b/bundles/org.openhab.ui/web/src/pages/developer/api-explorer.vue
@@ -115,44 +115,47 @@
   right 10px !important
 </style>
 
-<script>
-import auth from '@/components/auth-mixin.js'
+<script setup lang="ts">
+import { onMounted, ref } from 'vue'
+import { type Router } from 'framework7'
+import { getAccessToken, getTokenInCustomHeader, getBasicCredentials } from '@/js/openhab/auth'
 
-export default {
-  mixins: [auth],
-  props: {
-    f7router: Object
-  },
-  methods: {
-    onPageAfterIn () {
-      const swaggerCss = import(/* webpackChunkName: "swagger-css" */ 'swagger-ui-dist/swagger-ui.css')
-      const swaggerModule = import(/* webpackChunkName: "swagger" */ 'swagger-ui-dist')
-      const refreshToken = this.refreshAccessToken()
+import 'swagger-ui-dist/swagger-ui.css'
+import { SwaggerUIBundle, type SwaggerRequest } from 'swagger-ui-dist'
 
-      Promise.all([swaggerModule, swaggerCss, refreshToken]).then((results) => {
-        const SwaggerUI = results[0].SwaggerUIBundle
-        const tokenResponse = results[2]
-        SwaggerUI({
-          url: '/rest/spec',
-          dom_id: '#swaggerUi',
-          deepLinking: false,
-          defaultModelsExpandDepth: 0,
-          tagsSorter: 'alpha',
-          operationsSorter: 'alpha',
-          filter: true,
-          docExpansion: 'none',
-          syntaxHighlight: false,
-          requestInterceptor: (req) => {
-            if (document.cookie.indexOf('X-OPENHAB-AUTH-HEADER') >= 0) {
-              req.headers['X-OPENHAB-TOKEN'] = tokenResponse.access_token
-            } else {
-              req.headers['Authorization'] = 'Bearer ' + tokenResponse.access_token
-            }
-            return req
-          }
-        })
-      })
-    }
+// props and emits
+defineProps<{
+  f7router: Router.Router
+}>()
+
+// Events
+function onPageAfterIn() {
+  SwaggerUIBundle({
+    url: '/rest/spec',
+    dom_id: '#swaggerUi',
+    deepLinking: false,
+    defaultModelsExpandDepth: 0,
+    tagsSorter: 'alpha',
+    operationsSorter: 'alpha',
+    filter: true,
+    docExpansion: 'none',
+    syntaxHighlight: false,
+    requestInterceptor
+  })
+}
+
+// Methods
+function requestInterceptor(req: SwaggerRequest) {
+  const accessToken = getAccessToken()
+  if (getTokenInCustomHeader()) {
+    req.headers['X-OPENHAB-TOKEN'] = accessToken
+  } else {
+    req.headers['Authorization'] = 'Bearer ' + accessToken
   }
+  const basicCredentials = getBasicCredentials()
+  if (basicCredentials) {
+    req.headers['Authorization'] = 'Basic ' + btoa(basicCredentials.id + ':' + basicCredentials.password)
+  }
+  return req
 }
 </script>

--- a/bundles/org.openhab.ui/web/tsconfig.json
+++ b/bundles/org.openhab.ui/web/tsconfig.json
@@ -17,7 +17,7 @@
     "paths": {
       "@/*": ["src/*"]
     },
-    "types": ["audioworklet", "vite/client"]
+    "types": ["audioworklet", "vite/client", "swagger-ui-dist"]
   },
   "include": [
     "env.d.ts",

